### PR TITLE
Abstract the content of Dialog component into a new component named DialogContent

### PR DESCRIPTION
**Pull request by moyali**
#### Pull request checklist
- [x] Include a change request file if publishing
- [x] New feature, bugfix, or enhancement
- [x] Documentation update

#### Description of changes
This PR abstracts the content of the Modal inside the Dialog component into a new component named DialogContent. This separates the rendering of the dialog itself from the modal and enables creating a dialog without the OUFR Modal component. This is needed when the modal management and the rendering of the dialog content is managed by two separate entities.

### CODE REVIEW

**Comment by micahgodbolt** [comment]
Linting errors

File: packages/office-ui-fabric-react/src/components/Dialog/Dialog.Props.ts [hunk 1]
```diff
@@ -8,6 +9,40 @@ export interface IDialog {
...
+  * The subtext to display in the dialog.
+  */
+  subText?: string;
+
+  /**
+  * The title text to display at the top of the dialog.
+  */
+  title?: string;
+
+  id?: string;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Dialog/Dialog.Props.ts]: comment

File: packages/office-ui-fabric-react/src/components/Dialog/Dialog.Props.ts [hunk 2]
```diff
@@ -8,6 +9,40 @@ export interface IDialog {
...
+
+  /**
+  * Optional override class name
+  */
+  className?: string;
+
+  /**
+  * A callback function for when the Dialog is dismissed from the close button or light dismiss, before the animation completes.
+  */
+  onDismiss?: (ev?: React.MouseEvent<HTMLButtonElement>) => any;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Dialog/Dialog.Props.ts]: shouldn't onDismiss return `void`?

File: packages/office-ui-fabric-react/src/components/Dialog/Dialog.Props.ts [hunk 3]
```diff
@@ -8,6 +9,40 @@ export interface IDialog {
...
+
+  /**
+   * Other top buttons that will show up next to the close button
+   */
+  topButtonsProps?: IButtonProps[];
+
+  /**
+  * Optional override class name
+  */
+  className?: string;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Dialog/Dialog.Props.ts]: I'm not sure you're passing this in anywhere.

File: packages/office-ui-fabric-react/src/components/Dialog/Dialog.Props.ts [hunk 4]
```diff
@@ -8,6 +9,40 @@ export interface IDialog {
...
+export interface IDialogContentProps extends React.Props<DialogContent>, IWithResponsiveModeState, IAccessiblePopupProps {
+  /**
+  * Show an 'x' close button in the upper-right corner
+  */
+  showCloseButton?: boolean,
+
+  /**
+   * Other top buttons that will show up next to the close button
+   */
+  topButtonsProps?: IButtonProps[];
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Dialog/Dialog.Props.ts]: are you using this?

Comment (moyali) [inline packages/office-ui-fabric-react/src/components/Dialog/Dialog.Props.ts]: Fixed. Passed from Dialog.

File: packages/office-ui-fabric-react/src/components/Dialog/Dialog.Props.ts (hunk 3 above)
Comment (moyali) [inline packages/office-ui-fabric-react/src/components/Dialog/Dialog.Props.ts]: Fixed. Passed from Dialog.

File: packages/office-ui-fabric-react/src/components/Dialog/Dialog.Props.ts (hunk 1 above)
Comment (moyali) [inline packages/office-ui-fabric-react/src/components/Dialog/Dialog.Props.ts]: This prop should have been removed.

**Pull request merged**